spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -61,6 +61,12 @@
       expect(xmldoc.metadata_nodeset.length).to eq(1)
     end
 
+    it 'contains the correct NameID formats' do
+      # matching the spec, section 8.3
+      expected = { 'emailAddress' => '1.1', 'persistent' => '2.0' }
+      expect(xmldoc.metadata_name_id_formats).to eq(expected)
```

Comment:
I think I see what this test is doing, and I think the accompanying helper method makes it harder to read than intended.

I think a clearer test might look something like:

```
expect(name_id_version(xmldoc.email_address_name_id)).to eq('1.1')
expect(name_id_version(xmldoc.persistent_name_id)).to eq('2.0')
```

or maybe

```
expect(name_id_version(xmldoc.name_id_format('emailAddress')).to eq('1.1')
expect(name_id_version(xmldoc.name_id_format('persistent')).to eq('2.1')
```

so we'd define a `name_id_version(str)` and `XmlDoc#name_id_format(key)` methods

Reply:
That makes sense, I can definitely see how that improves readability. I'm on it 😄